Sources/ClientRuntime/Util/Async.swift

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Dispatch
+
+private class BoxedThrowingResult<ResultType> {
+    var result: Result<ResultType, Error>?
+}
+
+private class BoxedResult<ResultType> {
+    var result: ResultType?
+}
+
+/// Unsafely awaits a throwing async function from a synchronous context.
+@available(*, deprecated, message: """
+    This is a temporary workaround for the lack of async support.
+    It will be migrated to structured concurrency in the future
+""")
+public func _unsafeThrowingWait<ResultType>(_ f: @escaping () async throws -> ResultType) throws -> ResultType {
+    let box = BoxedThrowingResult<ResultType>()
+    let semaphore = DispatchSemaphore(value: 0)
+    Task {
+        do {
+            let val = try await f()
+            box.result = .success(val)
+        } catch {
+            box.result = .failure(error)
+        }
+        semaphore.signal()
+    }
+    semaphore.wait()
+    return try box.result!.get()
+}
+
+/// Unsafely awaits an async function from a synchronous context.
+@available(*, deprecated, message: """
+    This is a temporary workaround for the lack of async support.
+    It will be migrated to structured concurrency in the future
+""")
+public func _unsafeWait<ResultType>(_ f: @escaping () async -> ResultType) -> ResultType {
+    let box = BoxedResult<ResultType>()
+    let semaphore = DispatchSemaphore(value: 0)
+    Task {
+        box.result = await f()
+        semaphore.signal()
+    }
+    semaphore.wait()
+    return box.result!
```

Comment:
Can just use the `ResultType` here and remove `BoxedResult`? Similar to the above comment, imo this is simpler. 

```suggestion
    var result: ResultType? = nil
    let semaphore = DispatchSemaphore(value: 0)
    Task {
        result = await f()
        semaphore.signal()
    }
    semaphore.wait()
    return result!
```

Reply:
nope, same as above.